### PR TITLE
Enabled the taar_ensemble weekly job

### PR DESCRIPTION
**Pull request by crankycoder**
This creates a weekly dag and attaches the taar_ensemble job to it. 

I also added some extra documentation around the `pypi_libs` initializer argument to MozDatabricksSubmitOperator.

Parameters to the MozDatabricksSubmitRunOperator are from: https://github.com/mozilla/python_mozetl/pull/321#issuecomment-475828726

### CODE REVIEW

**Comment by crankycoder** [comment]
@acmiyaguchi r?  In particular - I wanted to make sure that I've done the right thing with respect to setting up a weekly dag in `main_summary.py`

File: dags/main_summary.py [hunk 1]
```diff
@@ -21,9 +21,23 @@
...
+    "retries": 2,
+    "retry_delay": timedelta(minutes=30),
+}
+
 # Make sure all the data for the given day has arrived before running.
 # Running at 1am should suffice.
 dag = DAG('main_summary', default_args=default_args, schedule_interval='0 1 * * *')
+dag_weekly = DAG(
+    "weekly_main_summary", default_args=default_args_weekly, schedule_interval="@weekly"
+)
```
Comment (acmiyaguchi) [inline dags/main_summary.py]: I'd probably move this out into a separate file like the hardware report: https://github.com/mozilla/telemetry-airflow/blob/master/dags/fx_usage_report.py

There's no hard dependency on the daily main_summary case here.

File: dags/main_summary.py [hunk 2]
```diff
@@ -642,5 +679,6 @@
 
 taar_locale_job.set_upstream(clients_daily_v6)
 taar_collaborative_recommender.set_upstream(clients_daily_v6)
+taar_ensemble.set_upstream(dag_weekly)
```
Comment (acmiyaguchi) [inline dags/main_summary.py]: The dag is already an argument to the operator.

**Comment by crankycoder** [comment]
@acmiyaguchi it's ok for the ensemble to be off by day - it's more important that that job run regularly rather than run on a precisely timed schedule.

File: dags/main_summary.py (hunk 2 above)
Comment (crankycoder) [reply dags/main_summary.py]: I've cleaned up the dag attribute assignment now in the constructor and set the upstream to be clients_daily.

File: dags/taar_weekly.py [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+
+
+
+dag_weekly = DAG(
+    "weekly_main_summary", default_args=default_args_weekly, schedule_interval="@weekly"
+)
+
+wait_for_clients_daily = ExternalTaskSensor(
+    task_id='clients_daily',
+    external_dag_id='clients_daily',
```
[the remark this answers is not in the document]
Comment (crankycoder) [reply dags/taar_weekly.py]: I've fixed the `external_dag_id` and `external_task_id` to be `main_summary` and `clients_daily` respectively.

File: dags/taar_weekly.py [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+    "email_on_failure": True,
+    "email_on_retry": True,
+    "retries": 2,
+    "retry_delay": timedelta(minutes=30),
+}
+
+
+
+dag_weekly = DAG(
+    "weekly_main_summary", default_args=default_args_weekly, schedule_interval="@weekly"
```
[the remark this answers is not in the document]
Comment (crankycoder) [reply dags/taar_weekly.py]: Renamed this dag to taar_weekly now.

**Comment by crankycoder** [comment]
I successfully started the task using `make run COMMAND="test taar_weekly taar_ensemble 20190601"`

```
--------------------------------------------------------------------------------
Starting attempt 1 of 1
--------------------------------------------------------------------------------

[2019-06-03 16:57:35,168] {models.py:1593} INFO - Executing <Task(MozDatabricksSubmitRunOperator): taar_ensemble> on 2019-06-01T00:00:00+00:00
[2019-06-03 16:57:35,194] {moz_databricks.py:239} INFO - Running TAAR Ensemble Model with parameters:
{'libraries': [{'pypi': {'package': u'git+https://github.com/mozilla/python_mozetl.git@master'}},
               {'pypi': {'package': u'mozilla-taar3==0.4.5'}},
               {'pypi': {'package': u'mozilla-srgutil==0.1.10'}},
               {'pypi': {'package': u'python-decouple==3.1'}}],
```

**Comment by crankycoder** [comment]
@acmiyaguchi thanks anthony.  i fixed those review comments up - and added a `start_date` argument as airflow was throwing an error about it when I tried to run under test.

File: dags/taar_weekly.py [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+    job_name="TAAR Ensemble Model",
+    owner="vng@mozilla.com",
+    email=["vng@mozilla.com", "mlopatka@mozilla.com"],
+    execution_timeout=timedelta(hours=11),
+    instance_count=5,
+    instance_type="i3.2xlarge",
+    spot_bid_price_percent=100,
+    max_instance_count=60,
+    enable_autoscale=True,
+    start_date='20190527',
```
Comment (acmiyaguchi) [inline dags/taar_weekly.py]: What's this keyword for?

Comment (crankycoder) [reply dags/taar_weekly.py]: https://github.com/apache/airflow/blob/master/airflow/models/baseoperator.py#L88

``` The ``start_date`` for the task, determines
        the ``execution_date`` for the first task instance.```

File: dags/taar_weekly.py [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+from airflow.operators.sensors import ExternalTaskSensor
+from datetime import datetime, timedelta
+from airflow.operators.moz_databricks import MozDatabricksSubmitRunOperator
+from utils.mozetl import mozetl_envvar
+
+default_args_weekly = {
+    "owner": "frank@mozilla.com",
+    "depends_on_past": False,
+    "start_date": datetime(2019, 5, 31),
+    "email": ["telemetry-alerts@mozilla.com", "frank@mozilla.com"],
```
[the remark this answers is not in the document]
Comment (crankycoder) [reply dags/taar_weekly.py]: removed in dad891c

**Pull request merged**